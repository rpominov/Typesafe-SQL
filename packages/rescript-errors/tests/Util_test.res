open Jest

let realWorld = `if i < start {
      highlightSymbols.contents->Js.Array2.push(_)->ignore
    } else if i <= end {
      highlightSymbols.contents->Js.Array2.push(_)->ignore
      highlightSymbolsEmpty := false
    }
`

each3(
  [("abc", 0, 0), ("abc", 3, 3), ("abc", 4, 4), (realWorld, 21, 95), (`>> ☀︎ <<`, 2, 4)],
  "%p %p %p",
  (code, start, end) => Util.highlight(~start, ~end, code)->expect->toMatchSnapshot,
)
