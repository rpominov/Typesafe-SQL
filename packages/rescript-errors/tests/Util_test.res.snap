// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`">> ☀︎ <<" 2 4 1`] = `
">> ☀︎ <<
  ^^^"
`;

exports[`"abc" 0 0 1`] = `
"abc
^"
`;

exports[`"abc" 3 3 1`] = `
"abc
   ^"
`;

exports[`"abc" 4 4 1`] = `
"abc
    ^"
`;

exports[`"if i < start {
      highlightSymbols.contents->Js.Array2.push(_)->ignore
    } else if i <= end {
      highlightSymbols.contents->Js.Array2.push(_)->ignore
      highlightSymbolsEmpty := false
    }
" 21 95 1`] = `
"if i < start {
      highlightSymbols.contents->Js.Array2.push(_)->ignore
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    } else if i <= end {
^^^^^^^^^^^^^^^^^^^^^^
      highlightSymbols.contents->Js.Array2.push(_)->ignore
      highlightSymbolsEmpty := false
    }
"
`;
